File: 11.container-with-most-water.java
// @leet start
class Solution {
  public int maxArea(int[] height) {
    int l = 0;
    int r = height.length - 1;
    int answer = 0;
    while (l < r) {
      answer = Math.max(answer, (r - l) * Math.min(height[l], height[r]));
      if (height[l] < height[r]) {
        l++;
      } else {
        r--;
      }
    }
    return answer;
  }
}
// @leet end

File: 11.container-with-most-water.py
# @leet start
class Solution:
    def maxArea(self, height: List[int]) -> int:
        Left = 0
        Right = len(height) - 1
        answer = 0
        while Left < Right:
            answer = max(answer, (Right - Left) * min(height[Left], height[Right]))
            if height[Left] < height[Right]:
                Left += 1
            else:
                Right -= 1
        return answer


# @leet end
